src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -245,16 +237,20 @@ private static boolean shouldBeOnSameLine(RightCurlyOption bracePolicy, Details
      * @param bracePolicy option for placing the right curly brace
      * @param details Details for validation
      * @return true if a right curly should be alone on a line.
+     * @noinspection OverlyComplexBooleanExpression
      */
+    // -@cs[BooleanExpressionComplexity] keeping logic in single method makes it clearer
```

Comment:
`&& bracePolicy == .... ` please move to the front and it become good.
Please apply this format of code.

> it is actually better to see all the logic in one place - it becomes clearer

:) yes, it is the always a reason when smb write a code, but it is disaster when you try to read it and understand